### ISSUE TITLE
Failed Pull Request Build on RTD shows as pending on GitHub

### ISSUE THREAD

**Issue by jakirkham** [readthedocs/readthedocs.org#8177]
Currently when a pull request build fails on RTD. It appears the GitHub status for that pull request is not updated as having failed, but remains marked as pending.

## Details

* GitHub PR: https://github.com/rapidsai/ucx-py/pull/734
* Read the Docs project URL: https://readthedocs.org/projects/ucx-py/
* Build URL (if applicable): https://readthedocs.org/projects/ucx-py/builds/13749560/
* Read the Docs username (if applicable):

## Expected Result

Once the RTD pull request build fails, it would be good to have the GitHub pull request status also marked as having failed.

## Actual Result

Here's a screenshot from RTD for the pull request. Note the build has failed

<img width="1792" alt="Screen Shot 2021-05-12 at 12 15 34 PM" src="https://user-images.githubusercontent.com/3019665/118031781-c412fe00-b31b-11eb-8c70-f3c79df6d97b.png">

Here's the screenshot of the GitHub pull request's CI statuses. Note RTD is shown as still pending

<img width="1792" alt="Screen Shot 2021-05-12 at 12 15 13 PM" src="https://user-images.githubusercontent.com/3019665/118031731-b8bfd280-b31b-11eb-849d-7c2c8c0afc19.png">

**Comment by jakirkham** [comment]
(apologies if this was in another issue somewhere. was unable to find it after searching. also feel free to move if this is not the right repo for this issue)

**Comment by humitos** [comment]
This may be a temporary failure when contacting GitHub. We could probably re-try the task up to 3 times if it fails at https://github.com/readthedocs/readthedocs.org/blob/f32ba4d4ca4e935b68bacf337cd88ac4ffdd12a4/readthedocs/projects/tasks.py#L1880 (it returns `False`)
